### ISSUE TITLE
Logging Broken

### ISSUE THREAD

**Issue by javierpra** [PowerShell/Win32-OpenSSH#1635]
**Troubleshooting steps**
https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Terminal issue? please go through wiki**
https://github.com/PowerShell/Win32-OpenSSH/wiki/TTY-PTY-support-in-Windows-OpenSSH

Please answer the following

**"OpenSSH for Windows" version**
 `((Get-Item (Get-Command sshd).Source).VersionInfo.FileVersion)`

 
**Server OperatingSystem**
 `((Get-ItemProperty "HKLM:\SOFTWARE\Microsoft\Windows nt\CurrentVersion\" -Name ProductName).ProductName)`
Windows Server 2012 R2 Standard

**Client OperatingSystem**
Windows Server 2012 R2 Standard

**What is failing**
The issue is with forwarding the OpenSSH event logs to a collector.  By default OpenSSH is creating a new folder in Event viewer called OpenSSH with the admin and operations logs under it.  A collector cannot see this folder and so is not able to subscribe to it.  Is there a possible work around for this?

**Expected output**

**Actual output**

**Comment by bagajjal** [comment]
@javierpra - Are you able to view the admin and operations logs in the event viewer?

**Comment by dwatley** [comment]
I'm having a similar issue, at least for sure in the latest release. Not able to view content via event viewer, as the logs are never populated.

![bad_logs](https://user-images.githubusercontent.com/2055181/87100297-e1a99f80-c200-11ea-80c6-3e0e0cdd8225.png)

On a working system it should be a single OpenSSH folder, with the Admin and Operational logs below it.

![working_logs](https://user-images.githubusercontent.com/2055181/87101328-aeb4db00-c203-11ea-9999-bac84ddf1d4d.png)

**Comment by bagajjal** [comment]
@dwatley - Can you try the below steps,
1) Uninstall existing Openssh version.
2) Install new V8.1 Openssh version which will create the ETW entries
3) Make sure the sshd_config doesn't have file based logging enabled.
4) Make a ssh connection 
5) Verify the logs in event viewer.

**Comment by dwatley** [comment]
During installation I get this error consistently which would appear to be related:

```
PS C:\Users\dwatley\Downloads\OpenSSH-Win64> .\install-sshd.ps1
**** Warning: Publisher OpenSSH resources could not be found or are not accessible
to the EventLog service account (NT SERVICE\EventLog).

[SC] SetServiceObjectSecurity SUCCESS
[SC] ChangeServiceConfig2 SUCCESS
[SC] ChangeServiceConfig2 SUCCESS
sshd and ssh-agent services successfully installed
PS C:\Users\dwatley\Downloads\OpenSSH-Win64>
```

This is straight from the downloaded zip, extracted, and running the install-sshd.ps1 file from an elevated PowerShell prompt.

**Comment by dwatley** [comment]
Stumbled across this [stackoverflow question](https://stackoverflow.com/questions/54324680/publisher-mytestsource-resources-could-not-be-found-or-are-not-accessible-to-the) which has an identical error. "Sharing" the extracted folder to `Everyone` and I don't get the error, and the event log is registered correctly.

So it seems to be a permissions issue, but it's not clear why.

**Comment by bagajjal** [comment]
@dwatley - Glad to know the issue is resolved.
@javierpra - Please check if this solves your problem.

**Comment by javierpra** [comment]
@bagajjal Thank you I will check and let you know.  Sorry for the late response I had new fires to put out.

**Comment by bpfoley45** [comment]
I too have run into this issue on a server 2019 system with OpenSSH 8.6p1.

The install-sshd.ps1 outputs the same error as @dwatley and while the event locations are created (in separate folders as pictured) nothing is logged.

I found that if you apply any ACL to C:\program files\OpenSSH that is too permissive (BUILTIN\users or Everyone); the sshd service will start but not allow connections. I am guessing there is some internal checking for sane configuration or overly permissive ACLs but haven't dug too far into this.

Looking at the error it seems that `NT Service\EventLog` is the permission needed, I tried applying this to the manifest file itself, but it failed as the account could not traverse the directory. So I granted read to the account to the parent directory and then installed the event manifest:

```
$acl = Get-Acl 'C:\Program Files\OpenSSH\'
$p = New-Object System.Security.Principal.NTAccount("NT Service\EventLog")
$accessRule = New-Object System.Security.AccessControl.FileSystemAccessRule($p, "Read", "ContainerInherit, ObjectInherit", "None", "Allow")
$acl.AddAccessRule($accessRule)
$acl | Set-Acl -Path 'C:\Program Files\OpenSSH\'
```


Now everything appears properly in Event Viewer and ssh connections are accepted.
